include/ua_securitycontext.h

```diff
@@ -0,0 +1,143 @@
+/*
+* Copyright (C) 2014 the contributors as stated in the AUTHORS file
+*
+* This file is part of open62541. open62541 is free software: you can
+* redistribute it and/or modify it under the terms of the GNU Lesser General
+* Public License, version 3 (as published by the Free Software Foundation) with
+* a static linking exception as stated in the LICENSE file provided with
+* open62541.
+*
+* open62541 is distributed in the hope that it will be useful, but WITHOUT ANY
+* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+* A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+* details.
+*/
+
+#ifndef UA_SECURITYCONTEXT_H_
+#define UA_SECURITYCONTEXT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_types.h"
+#include "ua_log.h"
+
+struct _UA_Channel_SecurityContext;
+typedef struct _UA_Channel_SecurityContext UA_Channel_SecurityContext;
+
+struct _UA_Policy_SecurityContext;
+typedef struct _UA_Policy_SecurityContext UA_Policy_SecurityContext;
+
+struct _UA_Policy_SecurityContext
+{
+    UA_StatusCode (*const init)(UA_Policy_SecurityContext* const securityContext, UA_Logger logger);
+
+    UA_StatusCode (*const deleteMembers)(UA_Policy_SecurityContext* const securityContext);
+
+    UA_StatusCode (*const setServerPrivateKey)(UA_Policy_SecurityContext* const securityContext,
+                                               const UA_ByteString* const privateKey);
+
+    UA_StatusCode (*const setCertificateTrustList)(UA_Policy_SecurityContext* const securityContext,
+                                                   const UA_ByteString* const trustList);
+
+    UA_StatusCode (*const setCertificateRevocationList)(UA_Policy_SecurityContext* const securityContext,
+                                                        const UA_ByteString* const revocationList);
+
+    void* data;
+
+    UA_Logger logger;
+};
+
+struct _UA_Channel_SecurityContext

```

Comment:
My initial thought was that the securityContext is its own object with its own data and methods (like in c++ for example) but i guess it would save a bit of memory if we didnt copy the function pointers for each channel. It should not be a problem to change this.
The memory needed for one Context should be around 100 Bytes with 64 bit pointers, i guess.